src/normalize-options.ts

```diff
@@ -76,10 +77,16 @@ export function normalizeOptions(
     ensured = true
   }
 
+  const optionsHash = stableHash(options)
   if (configFile) {
-    const cachedOptions = configFileMapping.get(configFile)
+    const cachedOptions = configFileMapping.get(`${configFile}\0${optionsHash}`)
     if (cachedOptions) {
-      log('using cached options for', configFile)
+      log(
+        'using cached options for',
+        configFile,
+        'with options hash',
```

Comment:
```suggestion
        'with options',
```

I don't think `options hash` make sense here, we should display `options` instead.

Reply:
769f6a6